setup.cfg

```diff
@@ -1,2 +1,71 @@
+[metadata]
+name = Lektor
+description = A static content management system.
+long_description = file: README.md
+long_description_content_type = text/markdown
+license = BSD
+platforms = any
+author = Armin Ronacher
+author_email = armin.ronacher@active-4.com
+url = http://github.com/lektor/lektor/
+classifiers =
+    Framework :: Lektor
+    Environment :: Web Environment
+    Intended Audience :: Developers
+    License :: OSI Approved :: BSD License
+    Operating System :: OS Independent
+    Programming Language :: Python
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.6
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+    Topic :: Internet :: WWW/HTTP :: Dynamic Content
+    Topic :: Software Development :: Libraries :: Python Modules
+requires = setuptools_scm
+
+[options]
+packages = find:
+include_package_data = True
+zip_safe = False
+python_requires = >=3.6
+install_requires =
+    Babel
+    click>=6.0
+    EXIFRead
+    filetype>=1.0.7
+    Flask
+    inifile
+    Jinja2>=3.0
+    mistune>=0.7.0,<2
+    pip
+    python-slugify
+    requests[security]
+    setuptools
+    watchdog
+    Werkzeug<3
+setup_requires =
+    # XXX: b/c: This duplicates config in pyproject.toml and is only
+    # used when running the stub setup.py directly
+    setuptools >= 45
+    setuptools_scm >= 6
+
+[options.extras_require]
+ipython =
+    ipython
+# XXX: the [test] extras is deprecated
+# Ref: https://github.com/lektor/lektor/pull/931#issuecomment-911879538
+test =
+    pre-commit
+    pylint==2.7.0
+    pytest
+    pytest-cov
+    pytest-mock
+    pytest-click
+
+[options.entry_points]
+console_scripts = 
+    lektor = lektor.cli:main
+
 [flake8]
```

Comment:
Should the `[flake8]` config be moved to `tox.ini`?

Flake8 will pick it up from either place.

It might be cleaner to keep the packaging config in `setup.cfg` and the test config in `tox.ini`.

Reply:
Sounds good

Reply:
I've now moved the flake8 config to tox.ini.

In so doing, I noticed that there were a lot of apparently extraneous ignored rules: some for rules from flake8 plugins which we don't currently use, some for rules which are not currently matched by our codebase (many of those are probably auto-fixed by the use of black.)

In a series of commits, I've cleaned up the unused ignores.